client.go

```diff
@@ -338,38 +338,43 @@ func (c *Client) Pull(ctx context.Context, ref string, opts ...RemoteOpt) (Image
 		}
 	}
 
-	imgrec := images.Image{
-		Name:   name,
-		Target: desc,
-		Labels: pullCtx.Labels,
-	}
-
-	is := c.ImageService()
-	if created, err := is.Create(ctx, imgrec); err != nil {
-		if !errdefs.IsAlreadyExists(err) {
-			return nil, err
-		}
-
-		updated, err := is.Update(ctx, imgrec)
-		if err != nil {
-			return nil, err
-		}
-
-		imgrec = updated
-	} else {
-		imgrec = created
-	}
-
 	img := &image{
 		client: c,
-		i:      imgrec,
+		i: images.Image{
+			Name:   name,
+			Target: desc,
+			Labels: pullCtx.Labels,
+		},
 	}
+
 	if pullCtx.Unpack {
 		if err := img.Unpack(ctx, pullCtx.Snapshotter); err != nil {
 			errors.Wrapf(err, "failed to unpack image on snapshotter %s", pullCtx.Snapshotter)
```

Comment:
`return nil, errors.Wrapf(err, "failed to unpack image on snapshotter %s", pullCtx.Snapshotter)`?

Reply:
Yeah, that isn't right https://github.com/containerd/containerd/pull/2011. I will make a seperate PR for that one.